torch/nn/parallel/distributed.py

```diff
@@ -1060,6 +1060,39 @@ def _register_comm_hook(self, state: object, hook: callable):
         self._check_comm_hook(hook)
         dist._register_comm_hook(self.reducer, state, hook)
 
+    def _register_builtin_comm_hook(
+        self, comm_hook_type: dist.BuiltinCommHookType
+    ):
+        r"""
+        Registers a built-in communication hook that specifies how DDP
+        aggregates gradients across multiple workers.
+
+        Arguments:
+            comm_hook_type (dist.BuiltinCommHookType): enum that corresponds
+                to a C++ comm hook implementation predefined in
+                torch/csrc/distributed/c10d/default_comm_hooks.h.
```

Comment:
This is implementation detail that we shouldn't mention in the docs here. You can add python docs to `BuiltinCommHookType` instead which would be available to python users.

Reply:
Thanks for catching this! Was not aware that this C++ code path is internal only.